error: redefinition of typedef ‘tagEntryInfo’

Unfortunately, recent revisions fail to build on my CentOS 6.5 system.  I used `git bisect` to determine that build failures began with revision e2755fded37efe060c8b260c07ecafaeaeb1d9e7.  Here's the error message emitted by GCC 4.4.7 when building e2755fded37efe060c8b260c07ecafaeaeb1d9e7:

    In file included from main/debug.h:20,
                     from main/args.c:20:
    main/entry.h:104: error: redefinition of typedef ‘tagEntryInfo’
    main/kind.h:23: note: previous declaration of ‘tagEntryInfo’ was here
    make: *** [main/args.o] Error 1
    make: *** Waiting for unfinished jobs....
    In file included from main/debug.h:20,
                     from main/entry.c:39:
    main/entry.h:104: error: redefinition of typedef ‘tagEntryInfo’
    main/kind.h:23: note: previous declaration of ‘tagEntryInfo’ was here

I get the same error message when building master (currently 8009da7753b177fdc56d3284f71b5f63178a362a), except for different line numbers.

I'm not sure how to fix this, so I'm hoping that this is simple enough for someone else to fix.

Thanks for developing Universal Ctags!

--- a/main/field.c
+++ b/main/field.c
@@ -383,7 +383,7 @@ static const char *renderFieldRole (const tagEntryInfo *const tag, vString* b)
 	{
 		Assert (rindex < tag->kind->nRoles);
 		role  = & (tag->kind->roles [rindex]);
-		return renderRole (role, tag, b);
+		return renderRole (role, b);
 	}
 
 	return vStringValue (b);

--- a/main/kind.c
+++ b/main/kind.c
@@ -21,8 +21,7 @@ extern void printRole (const roleDesc* const role)
 		printf ("%s\t%s\t%s\n", role->name, role->description, role->enabled? "on": "off");
 }
 
-extern const char *renderRole (const roleDesc* const role,
-			       const tagEntryInfo *const tag __unused__, vString* b)
+extern const char *renderRole (const roleDesc* const role, vString* b)
 {
 	vStringCatS (b, role->name);
 	return vStringValue (b);

--- a/main/kind.h
+++ b/main/kind.h
@@ -19,10 +19,8 @@ typedef struct sRoleDesc {
 	const char* description;	  /* displayed in --help output */
 } roleDesc;
 
-struct sTagEntryInfo;
-typedef struct sTagEntryInfo tagEntryInfo;
 extern void printRole (const roleDesc* const role); /* for --help */
-extern const char *renderRole (const roleDesc* const role, const tagEntryInfo *const tag, vString* b);
+extern const char *renderRole (const roleDesc* const role, vString* b);
 
 /*
  * Predefined kinds

